### ISSUE TITLE
It is allowed to sign up with the same email several times when there are caps variations

### ISSUE THREAD

**Issue by luisg123v** [odoo/odoo#25532]
Impacted versions:
- 11.0

Steps to reproduce:
- Enable free sign up: go to Settings ->  General Settings, and check "Free sign up (B2C)"
- Log out
- Go to the sign up form by clicking "Don't have an account?" at the log in page
- Fill the form with the email john1@example.com and whatever in the name and password
- Click Sign up
- Log out and sign up again with the email john1@EXAMPLE.com

Current behavior:
A new user is created with the email john1@EXAMPLE.com.

Expected behavior:
Since it's actually the same email, the following error message should come up:
> Another user is already registered using this email address.

**Comment by luisg123v** [comment]
CC @yaniaular

**Comment by pedrobaeza** [comment]
I think has been discussed other times, isn't it, @mart-e ?

**Comment by mart-e** [comment]
Indeed, the login is currently case sensitive and it would be good to change it in the future.
We already talked about changing this in the future and adding a `strip()` on the login with @odony. It is a bit tricky, the chance of breaking some authentication modules is quite high.

**Comment by luisg123v** [comment]
@mart-e 
Couldn't be possible to change [this line in auth_signup](https://github.com/odoo/odoo/blob/8643175d00fb8ac418af15273cdbce32f9946c50/addons/auth_signup/controllers/main.py#L50), to replace `=` by `=ilike`? That shouldn't have any negative impact, but it would make possible  for an external module to handle this via a constraint in `res.users`, without needing to replace the entire method.

I'll understand if such change is not possible, of course.

**Comment by odony** [comment]
@luisg123v I don't think that would work to fix your problem, this line is only used to select the error message. In you case there is no error, both logins are considered different.

More generally, we can't change login case-sensitivity without causing major compatibility problems. What do we do when conflicting logins do exist and are already in use?
Additionally, there is no official standard saying that mailbox names are case-insensitive. Some email providers could decide that "bg@domain.com" is a different mailbox/user than "BG@domain.com". Even though that is weird and unusual, it's allowed by the standard ([RFC 5321](https://tools.ietf.org/html/rfc5321#section-2.4)).

For more context info, have a look at this [discussion](https://stackoverflow.com/questions/9807909/are-email-addresses-case-sensitive). You will notice in the comments that some users confirm that their organization uses MS Exchange and some colleagues have mailboxes that only differ by case! :open_mouth:

**Comment by moylop260** [comment]
@luisg123v 
Did you mean, we should use a filter like as
```python
user, domain = qcontext.get("login").split('@')
request.env["res.users"].sudo().search([("login", "=like", user + '@%'), ("login", "=ilike",  '%@' + domain)])
```
?

**Comment by luisg123v** [comment]
@moylop260 
Yeah, that's a good alternative. However, the constraint is actually located [here](https://github.com/odoo/odoo/blob/4d0a1330e05bd688265bea14df4ad12838f9f2d7/odoo/addons/base/res/res_users.py#L236). The filter is used only to determine whether the cause of a sign-up  failure is a duplicated login, so '=' might be enough? That way, the constraint on `res.users` could be overridden by a local module to evaluate emails in a case-insensitive manner, and auth_signup will catch it correctly.

**Comment by odony** [comment]
> Notice that on this issue I made variations on the domain, not on the user name; wich, accordin to RFC 1035, should not be compared sensitively.

Of course, but the problem and its solution apply to the whole login, not just the domain. And there is no guarantee that a login *is* an email adress, either.

**Comment by C3POdoo** [comment]
Dear @luisg123v,

_This is an automated message._
This issue is linked to the Odoo support ticket opw-1862906 that has been closed.
We hope you got an answer to your issue.

If you believe the problem is not solved yet and would like to keep this issue open, please let us know to reopen it.
